### PR TITLE
Unified api adds integrations

### PR DESCRIPTION
**Pull request by fetzig**

### CODE REVIEW

File: Sources/Sentry/SentryHub.m [hunk 1]
```diff
@@ -100,4 +105,20 @@ - (void)configureScope:(void(^)(SentryScope *scope))callback {
     }
 }
 
+- (BOOL)doInstallIntegrations {
+    // TODO(fetzig) change this. instead of skipping whenever integrations have already been installed, check the integrations one by one, and skip those that are already installed.
+    if (SentrySDK.currentHub.installedIntegrations.count > 0) {
+        [SentryLog logWithMessage:@"[SentryHub doInstallIntegrations] there are already installed integrations. skipping isntall." andLevel:kSentryLogLevelError];
+        return NO;
+    }
```
Comment (HazAT) [inline Sources/Sentry/SentryHub.m]: Can we check here for every specific integration? This is important.

File: Sources/Sentry/SentryClient.m [hunk 2]
```diff
@@ -74,18 +69,15 @@ @implementation SentryClient
 - (_Nullable instancetype)initWithOptions:(SentryOptions *)options
                          didFailWithError:(NSError *_Nullable *_Nullable)error {
```
Comment (HazAT) [inline Sources/Sentry/SentryClient.m]: Can we just remove the `didFailWithError` part? This is no longer used here. Also removes the check later in the static API

File: Sources/Sentry/SentryHub.m (hunk 1 above)
Comment (fetzig) [reply Sources/Sentry/SentryHub.m]: perfect. wasn't sure. done.

File: Sources/Sentry/SentryClient.m (hunk 2 above)
Comment (fetzig) [reply Sources/Sentry/SentryClient.m]: done

File: Sources/Sentry/SentryCrashIntegration.m [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+                                       language:language
+                                     lineOfCode:lineOfCode
+                                     stackTrace:stackTrace
+                                  logAllThreads:logAllThreads
+                               terminateProgram:terminateProgram];
+    [installation sendAllReports];
+}
+
+// TODO(fetzig) this was in client, used for testing only, not sure if we can still use this (for testing). maybe move it to hub or static-sdk?
+- (void)snapshotStacktrace:(void (^)(void))snapshotCompleted {
```
Comment (HazAT) [inline Sources/Sentry/SentryCrashIntegration.m]: We can remove this function

File: Sources/Sentry/SentryCrashIntegration.m [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+        installation = [[SentryInstallation alloc] init];
+        [installation install];
+        [installation sendAllReports];
+    });
+    return YES;
+}
+#pragma GCC diagnostic pop
+
+// TODO(fetzig) this was in client, used for testing only, not sure if we can still use this (for testing). maybe move it to hub or static-sdk?
+- (void)reportUserException:(NSString *)name
```
Comment (HazAT) [inline Sources/Sentry/SentryCrashIntegration.m]: We can remove this function

File: Sources/Sentry/SentryIntegrationProtocol.h [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+/**
+ * installs the integration and returns YES if no error(s) occured
+ * NOTE: SentryOptions argument isn't always needed by the actual class implementing this protocol.
+ */
+- (BOOL)installWithOptions:(SentryOptions *)options;
+
+/**
+ * version of integration
+ */
+- (NSString *)version;
```
Comment (HazAT) [inline Sources/Sentry/SentryIntegrationProtocol.h]: We can remove this, as discussed.

File: Sources/Sentry/SentryIntegrationProtocol.h [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+ * name of integration
+ * class name in most cases
+ */
+- (NSString *)name;
+
+/**
+ * combination of name and version.
+ * supposed to be attached to the event so event processors to identify event data format.
+ */
+- (NSString *)identifier;
```
Comment (HazAT) [inline Sources/Sentry/SentryIntegrationProtocol.h]: We can remove this, as discussed.

**Pull request merged**